### ISSUE TITLE
Request notification of Adventure Curator (creator) of modifications to adventure(s) by other Curators.

### ISSUE THREAD

**Issue by NorNamlin** [AdventureLookup/AdventureLookup#237]
I would like to see a notification sent to the curator of an adventure that shows what changes have been made to their adventure.

This may help them to understand the looked for format or other changes recommended. This way they do not make the same possible mistake on further adventure submits.

This also allows for a review by the adventure creator to review changes and ensure that they are in fact correct.

I have found changes to my submitted adventures, that while most make sense or are needed, are not all always correct.

**Comment by burnhamrobertp** [comment]
Presently, changes are totally disconnected from change requests. The easiest way for this to work would be to notify any submitter of an open pull request anytime the corresponding adventure was edited... But that's not actually good and I wouldn't encourage anyone to build that. It's riddled with tricky exceptions, for example, multiple open change requests or what if the cursor resolved the change request before making the edit?

**Comment by burnhamrobertp** [comment]
It's a good idea but we'd need more work connecting a specific edit to a specific change request

**Comment by NorNamlin** [comment]
Interesting.

The issue of course, changes are made to an entry.  It goes unnoticed by the creator, but has been made in error.

 At what point will this get corrected?  There are no flags indicating a change to an entry as you note above, they are disconnected.  I agree, you don't want both to be available as it will cause conflicts.  But short of checking the submissions by the Creator at a regular interval, changes can be made that my or may not be correct.

 All I am asking, is for a change that is made to be reported in some way to the Creator, so they can review the change by a curator to check it's validity and ensure that both are following the spirit of the entry per Matt's guidelines.

**Comment by MattColville** [comment]
Here's how I imagined it working.

Someone submits an adventure. Some of their data is wrong.

Someone makes changes to that record. Some of those changes are good, some are bad.

Someone runs the adventure, discovers that some of the entries are bad, and goes in and fixes them. This may introduce more errors.

But, over time, the database is generally becoming more correct. 

Entries don't have owners. I'm not even sure how they could have owners. I'm not sure what ownership would mean. All we can do is count on the database to improve over time.

**Comment by NorNamlin** [comment]
I like the idea behind that approach.

The only thing I see is the "owners."  I would think that the person entering the initial data is the "creator" and they, at this current time, have the ability to go back and edit the entry in any way they see fit or as requested in a change log if the information is deemed to be correct.  This implies ownership of the entry.

Does that make sense?  I see as to why there may be arguments made that they do or do not have an owner.  But the creator can entirely delete an entry at any given time currently.  This also implies a sort of ownership to the entry.

I am not sure that creators of the entries they submit understand how this issue is being handled.  I know I certainly did not.  I did not think of myself as the owner per say, but to an extent, the person responsible for keeping the entry as accurate as possible.

Short of submitting a change request for data they find to be incorrect, they can not directly make a change to an entry like a curator can.

All I was indicating is that it would be a good thing, I would think, to have a notification to the creator of the entry to be made aware of changes to entries they have submitted done by a curator so they may be reviewed by the creator to ensure they understand why the change was met and that all information pertaining to the correction is correct.

An example of this is https://www.adventurelookup.com/adventures/dire-omens.  I submitted this entry.  A curator made mention of the Boss Monster entries I have.  I asked what needs changed and the curator opted to make the changes on their own to meet what they think are the guidelines for the Boss Monster field.  I have since gone back and re entered them as I had originally done as I feel that it is more in line with how you see the field being populated.

Had I not gone back and reviewed this entry due to the change request no longer being shown as needed on it, I would have had no idea that the curator made the changes that I deem wrong and corrected them.

So I would say, that over time, the database will continue to have errors unrealized as time goes on.

**Comment by burnhamrobertp** [comment]
That you have greater control over an adventurer than someone who didn't submit it does imply an ownership that I'm "on the fence" about.

I could see ownership being an actual thing, with Kobold Press actually signing up with an account and entering their adventures (or "claiming" the adventures that they had published that hadn't already been claimed by their specific authors, for example). This could lighten the load on curators but has its own problems associated with it.

Ultimately, AL doesn't support it right now so its somewhat moot. That you have greater control over the adventure than every other person who didn't submit it may not be desirable. It creates the exact situation you describe where a CR went through and a curator decided to make that change and you - as the original submitter - literally undid it on your own because you disagreed with the change.

This partially comes down to preference and interpretation of the intent of the data. To me, the boss monsters for the adventure you listed are pretty useless:
![image](https://user-images.githubusercontent.com/231052/31210843-087fc3aa-a963-11e7-9f97-b855c8371cd4.png)

If that were "Lich, Ghoul, Mummy, Ghost" I'd have the exact same amount of useful information than what it does now. But, I'd also have the ability to use the boss monster filter with "Lich" selected and see this adventure in the results. I'd also not have 4 unique boss monsters cluttering up the boss monsters list which creates a maintenance cost for the system, increases developmental costs, and complicates implementation of anything and everything that is connected to boss monsters.

This ties into me creating the ticket #236 , but I think I've phrased it better here.

"Lots of data" is not free. The common monsters, boss monsters, and notable items being 1973, 950, and  1938 items long, respectively, become problems in and of themselves that must be accounted for when interfaces are designed, UX is considered for certain screens / pages, or when action is taken. It complicates behaviors for users, curators, and admins. It is most costly for new features, for example. If we add `species` as a data type that's connected to monsters just like `type` is now (as suggested by Matt's last comment in #137), how easy is that going to be? How will it affect other areas of the system?

**Comment by NorNamlin** [comment]
I agree about "if" someone owns the entry created.  I would also agree that if a company, such as Kobold Press you indicated, wished to sign up and enter their own adventures, they certainly should claim ownership of their property.  Though that does not address the other problems you noted.

I also agree that you would get the results you are looking for if you chose "Lich, Ghoul, Mummy and Ghost" in the filter, you would get useful information for what you are looking for.  But what about others who are looking for additional adventures with say "Bargle" in them?  Which format do you use to find these entries?  Humanoid certainly wouldn't be appropriate would it?

So unless unique entries are going to require approval before they are entered into the data base, maybe by field, not the entire adventure entry itself, there will never be a resolution to these issues that are easing up "lots of data."
